src/Lykke.Bil2.RabbitMq/Subscription/Core/RetryManager.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using C5;
+using Common.Log;
+using Lykke.Common.Log;
+using Nito.AsyncEx;
+
+namespace Lykke.Bil2.RabbitMq.Subscription.Core
+{
+    internal class RetryManager : BackgroundService, IRetryManager
+    {
+        private readonly ILog _log;
+        private readonly TimeSpan _maxAgeForRetry;
+        private readonly int _maxRetryCount;
+        private readonly AsyncLock _mutex;
+        private readonly IPriorityQueue<(DateTime, EnvelopedMessage)> _retryQueue;
+        private readonly IInternalMessageQueue _internalQueue;
+
+        public RetryManager(
+            ILogFactory logFactory,
+            TimeSpan maxAgeForRetry,
+            int maxRetryCount,
```

Comment:
Limit queue size. If queue is full, the message should be rejected to the RabbitMq immediatelly.

Reply:
Fixed